llvm/lib/CodeGen/TargetInstrInfo.cpp

```diff
@@ -1414,7 +1418,7 @@ void TargetInstrInfo::reassociateOps(
     if (Idx == 0)
       continue;
     if (Idx == RootFirstOpIdx)
-      MIB2 = MIB2.addReg(RegA, getKillRegState(KillA));
+      MIB2 = MIB2.addReg(RegA, getKillRegState(KillA), SubRegA);
```

Comment:
I think you might be missing a swap in the SwapRootOperands just above.  Not sure what a sensible subreg is for NewVR is though.

Reply:
It's used as a result register, and this is still SSA so it cannot have a subregister